### DISCUSSION TITLE
Linux test results

### DISCUSSION THREAD

**Discussion by Vort** [bochs-emu/Bochs#94] [Show and tell]
Hello.

After seeing request for Linux testing [here](https://www.fysnet.net/bochs/images/supported.htm), I decided to try it with latest version of Bochs (221cac79720387b33a84c915a922af3c9527f04e).

First attempt was with **SliTaz** (`slitaz-rolling-core64.iso`).
After ~30 minutes of unpacking, it requested login and password _(strange idea for LiveCD)_.
But because of default clock settings of Bochs, I was not able to enter login correctly.

Second attempt was with **Puppy Linux** (`fossapup64-9.5.iso`).
After ~1 hour of loading black screen appeared and nothing happened within the next 30 minutes.
Probably it failed to switch graphical mode and also failed to notify user about it.

Third attempt was with **Tiny Core** (`TinyCore-current.iso`).
It successfully loaded within ~5 minutes:
![bochs_tinycore_14](https://github.com/bochs-emu/Bochs/assets/1242858/a5133fe0-2500-4f2c-992c-170fd43c9be0)
USB mouse and keyboard was detected correctly and was working correctly:
![bochs_tinycore_14_usb](https://github.com/bochs-emu/Bochs/assets/1242858/57e04421-8ea2-46f3-af92-bc2b10964492)
Looks like this OS variation suits well for USB tests.

**Comment by stlintel** [comment]
I see on screenshot the IPS is ~75M
What is your ips setting in .bochsrc ?
It should be around 50M-150M
The farther you go from true 75M the worse result you supposed to get

I'll try them here later and report

**Comment by stlintel** [comment]
Can you add full download link for SliTaz (slitaz-rolling-core64.iso) and Puppy Linux (fossapup64-9.5.iso) ?

**Comment by Vort** [comment]
> I see on screenshot the IPS is ~75M
> What is your ips setting in .bochsrc ?

`cpu: count=1:1:1, ips=50000000, quantum=16, model=core2_penryn_t9600, reset_on_triple_fault=1, cpuid_limit_winnt=0, ignore_bad_msrs=1, mwait_is_nop=0, msrs="msrs.def"`
`clock: sync=slowdown, time0=local, rtc_sync=0`

Without `slowdown` I expect to have problems with keyboard repeating characters very fast.
For example, for failed attempt with SliTaz (without `slowdown`), there was around 500M IPS at login screen.

> Can you add full download link for SliTaz (slitaz-rolling-core64.iso) and Puppy Linux (fossapup64-9.5.iso) ?

http://mirror.slitaz.org/iso/rolling/slitaz-rolling-core64.iso
https://distro.ibiblio.org/puppylinux/puppy-fossa/fossapup64-9.5.iso

**Comment by stlintel** [comment]
Slitaz booted perfectly to GUI within 3-5 minutes. Then I opened terminal and it looked like all good.

With the following CPU setting:

```
cpu: model=corei7_icelake_u
cpu: count=1, ips=500M, reset_on_triple_fault=1, ignore_bad_msrs=1
cpu: cpuid_limit_winnt=0

```
I use 2GB memory setting: 
memory: guest=2048, host=2048

And my host is Core I7 12900 which is capable to reach 200M-250M IPS natively.

**Comment by stlintel** [comment]
Puppy Linux (fossapup64-9.5.iso): about 10 minutes to boot

![image](https://github.com/bochs-emu/Bochs/assets/77239907/d13ded24-9b2f-48e3-91d6-99f72fe0d3b6)

I use cirrus graphics in configuration

**Comment by stlintel** [comment]
my .bochsrc for reference
[bochsrc.cdrom.txt](https://github.com/bochs-emu/Bochs/files/12921780/bochsrc.cdrom.txt)

**Comment by Vort** [comment]
After rebuilding Bochs with proper optimization, I tried to run SliTaz again.
I configured `ips=200000000` and my CPU was able to get 160M IPS during decompression.
With `sync=slowdown` at login screen IPS was 300M, but I was able to enter login (`tux`) successfully.
GUI of SliTaz did not loaded however, so I tried to enable cirrus video next time.
After that, GUI loaded, but (USB) mouse cursor was very laggy - movements, which took half second of real time, were executed by OS during 4 seconds.
Similar problems happened with keyboard.
Do not know if I made some mistakes with configuration or it's a sign of problems with Bochs.
Here is a screenshot of loaded OS:
![bochs-slitaz](https://github.com/bochs-emu/Bochs/assets/1242858/1aa85f40-d33c-4495-b33a-7e806135f2e8)

**Comment by Vort** [comment]
After enabling cirrus video, Puppy Linux loaded successfully.
Also I have found why lags were happening:
I was using `chipset=i440bx` instead of `chipset=i440fx` for enabling cirrus video.
